e2e/test/scaling/provider.go

```diff
@@ -39,42 +44,56 @@ func NewProvider(config ProviderConfig) (*Provider, error) {
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
-
 	if config.ClusterID == "" {
 		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
 	}
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
 
 	p := &Provider{
 		guestFramework: config.GuestFramework,
 		hostFramework:  config.HostFramework,
 		logger:         config.Logger,
 
-		clusterID: config.ClusterID,
+		clusterID:  config.ClusterID,
+		ctrlClient: config.CtrlClient,
 	}
 
 	return p, nil
 }
 
-func (p *Provider) AddWorker() error {
-	customObject, err := p.hostFramework.G8sClient().ProviderV1alpha1().AzureConfigs("default").Get(p.clusterID, metav1.GetOptions{})
-	if err != nil {
-		return microerror.Mask(err)
+func (p *Provider) findMachinePool(ctx context.Context) (*v1alpha3.MachinePool, error) {
+	crs := &v1alpha3.MachinePoolList{}
+
+	var labelSelector client.MatchingLabels
+	{
+		labelSelector = make(map[string]string)
+		labelSelector[capiv1alpha3.ClusterLabelName] = p.clusterID
 	}
 
-	patches := []Patch{
-		{
-			Op:    "add",
-			Path:  "/spec/azure/workers/-",
-			Value: customObject.Spec.Azure.Workers[0],
-		},
+	err := p.ctrlClient.List(ctx, crs, labelSelector, client.InNamespace(metav1.NamespaceDefault))
+	if err != nil {
+		return &v1alpha3.MachinePool{}, microerror.Mask(err)
+	}
+	if len(crs.Items) < 1 {
+		p.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("MachinePool CR for cluster id %q not found", p.clusterID))
+		return &v1alpha3.MachinePool{}, microerror.Maskf(notFoundError, fmt.Sprintf("MachinePool CR for cluster id %q not found", p.clusterID))
 	}
 
-	b, err := json.Marshal(patches)
+	return &crs.Items[0], nil
+}
+
+func (p *Provider) AddWorker() error {
+	ctx := context.TODO()
```

Comment:
```suggestion
	ctx := context.Background()
```
https://github.com/giantswarm/fmt/blob/master/go/context.md#guidelines